hazelcast/src/hazelcast/client/network.cpp

```diff
@@ -64,1271 +64,1639 @@
+    }
+}
 
-            void ClientConnectionManagerImpl::check_partition_count(int32_t new_partition_count) {
-                auto &partition_service = static_cast<spi::impl::ClientPartitionServiceImpl &>(client_.get_partition_service());
-                if (!partition_service.check_and_set_partition_count(new_partition_count)) {
-                    BOOST_THROW_EXCEPTION(exception::client_not_allowed_in_cluster("ClientConnectionManagerImpl::check_partition_count",
-                          (boost::format("Client can not work with this cluster because it has a different partition count. "
-                                         "Expected partition count: %1%, Member partition count: %2%")
-                                         %partition_service.get_partition_count() %new_partition_count).str()));
-                }
-            }
+std::shared_ptr<Connection>
+ClientConnectionManagerImpl::on_authenticated(
+  const std::shared_ptr<Connection>& connection,
+  auth_response& response)
+{
+    {
+        std::lock_guard<std::recursive_mutex> guard(client_state_mutex_);
+        check_partition_count(response.partition_count);
+        connection->set_connected_server_version(response.server_version);
+        connection->set_remote_address(response.server_address);
+        connection->set_remote_uuid(response.member_uuid);
+
+        auto existing_connection =
+          active_connections_.get(response.member_uuid);
+        if (existing_connection) {
+            connection->close(
+              (boost::format(
+                 "Duplicate connection to same member with uuid : %1%") %
+               boost::uuids::to_string(response.member_uuid))
+                .str());
+            return existing_connection;
+        }
 
-            void ClientConnectionManagerImpl::trigger_cluster_reconnection() {
-                if (reconnect_mode_ == config::client_connection_strategy_config::reconnect_mode::OFF) {
-                    HZ_LOG(logger_, info, "RECONNECT MODE is off. Shutting down the client.");
-                    shutdown_with_external_thread(client_.get_hazelcast_client_implementation());
-                    return;
-                }
+        auto new_cluster_id = response.cluster_id;
+        boost::uuids::uuid current_cluster_id = cluster_id_;
+
+        HZ_LOG(logger_,
+               finest,
+               boost::str(boost::format(
+                            "Checking the cluster: %1%, current cluster: %2%") %
+                          new_cluster_id % current_cluster_id));
+
+        auto cluster_id_changed = !current_cluster_id.is_nil() &&
+                                  !(new_cluster_id == current_cluster_id);
+        if (cluster_id_changed) {
+            HZ_LOG(
+              logger_,
+              warning,
+              boost::str(
+                boost::format(
+                  "Switching from current cluster: %1%  to new cluster: %2%") %
+                current_cluster_id % new_cluster_id));
```

Comment:
Do we really want this log line span these lines?

Reply:
If it is longer than 80 characters, yes.